base/ltpictur.dtx

```diff
@@ -237,34 +293,98 @@
 % changed 20 Jul 87).
 %
 %    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2020/10/01}%
+%<latexrelease>                 {\put}{default units}%
+%<latexrelease>\expandafter\let\csname put \endcsname\@undefind
 \long\def\put(#1,#2)#3{%
-  \@killglue\raise#2\unitlength
-  \hb@xt@\z@{\kern#1\unitlength #3\hss}%
+  \@killglue
+  \@defaultunitsset\@tempdimc{#2}\unitlength
+  \raise\@tempdimc
+  \hb@xt@\z@{%
+    \@defaultunitsset\@tempdimc{#1}\unitlength
+    \kern\@tempdimc
+    #3\hss}%
   \ignorespaces}
+%</2ekernel|latexrelease>
+%    \end{macrocode}
+%    
+%    \begin{macrocode}
+%<latexrelease>\EndIncludeInRelease
+%<latexrelease>\IncludeInRelease{0000/00/00}%
+%<latexrelease>                 {\put}{default units}%
+%<latexrelease>\expandafter\let\csname put \endcsname\@undefind
+%<latexrelease>\long\def\put(#1,#2)#3{%
+%<latexrelease>  \@killglue\raise#2\unitlength
+%<latexrelease>  \hb@xt@\z@{\kern#1\unitlength #3\hss}%
+%<latexrelease>  \ignorespaces}
+%<latexrelease>\EndIncludeInRelease
+%<*2ekernel>
 %    \end{macrocode}
 %
 %
 % \begin{macro}{\multiput}
 % \changes{v0.1c}{1994/04/28}{(DPC) Ignore spaces between )(}
 % |#3| had better be a |(|.
 %    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2020/10/01}%
+%<latexrelease>                 {\multiput}{default units}%
+%<latexrelease>\expandafter\let\csname multiput \endcsname\@undefind
 \def\multiput(#1,#2)#3{%
-  \@xdim #1\unitlength
-  \@ydim #2\unitlength
+  \@defaultunitsset\@xdim{#1}\unitlength
+  \@defaultunitsset\@ydim{#2}\unitlength
    \@multiput(}
+%</2ekernel|latexrelease>
+%    \end{macrocode}
+%    
+%    \begin{macrocode}
+%<latexrelease>\EndIncludeInRelease
+%<latexrelease>\IncludeInRelease{0000/00/00}%
+%<latexrelease>                 {\multiput}{default units}%
+%<latexrelease>\expandafter\let\csname multiput \endcsname\@undefind
+%<latexrelease>\def\multiput(#1,#2)#3{%
+%<latexrelease>  \@xdim #1\unitlength
+%<latexrelease>  \@ydim #2\unitlength
+%<latexrelease>   \@multiput(}
+%<latexrelease>\EndIncludeInRelease
+%<*2ekernel>
 %    \end{macrocode}
 % \end{macro}
 %
-% \begin{macro}{\multiput}
+% \begin{macro}{\@multiput}
 % \changes{v0.1c}{1994/04/28}{(DPC) Macro added}
 %    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2020/10/01}%
+%<latexrelease>                 {\@multiput}{default units}%
 \long\def\@multiput(#1,#2)#3#4{%
   \@killglue\@multicnt #3\relax
   \@whilenum \@multicnt >\z@\do
     {\raise\@ydim\hb@xt@\z@{\kern\@xdim #4\hss}%
      \advance\@multicnt\m@ne
-     \advance\@xdim#1\unitlength\advance\@ydim#2\unitlength}%
+     \@defaultunitsset{\advance\@xdim}{#1}\unitlength
+     \@defaultunitsset{\advance\@ydim}{#2}\unitlength}%
```

Comment:
what that usage documented in the definition?

Reply:
I intended to document it but haven't yet it seems. The alternative is to document that only a register is used as `#1` and do this in two steps, default units to `\@tempdimc` then add that to `\@xdim`  but for an internal call this seems neater.

Reply:
documented use of `\advance`